### PR TITLE
Fixed bug with showing all map layers and map collections (DI2-209)

### PR DESCRIPTION
**Pull request by alimpens**

### CODE REVIEW

**Review by chrisvanmook — APPROVED** [review]

**Review by jonkoops — APPROVED** [review]

File: src/store/middleware/preserveUrlParametersMiddleware.test.ts [hunk 1]
```diff
@@ -26,7 +26,19 @@ describe('Custom Redux Middleware', () => {
     expect(getParametersForRouteMock).not.toHaveBeenCalled()
   })
 
-  it('should use the custom middleware when pathname does not includes "kaart"', () => {
+  it('should use the custom middleware when pathname includes "kaarten"', () => {
+    // @ts-ignore
+    window.location = {
```
Comment (chrisvanmook) [inline src/store/middleware/preserveUrlParametersMiddleware.test.ts:31]: can't you do:
Object.defineProperty(window, 'location', {
    value: {
      ...window.location,
      pathname: '/kaarten/foo/bar',
      search: '?foo=bar',
    },
  })

**Review by chrisvanmook — APPROVED** [review]

**Review by chrisvanmook — APPROVED** [review]

File: src/app/pages/MapPage/detail/buildDetailUrl.test.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import buildDetailUrl from './buildDetailUrl'
+
+describe('return detail panel path', () => {
+  it('should return a path starting with "/kaart/" when pathname includes "kaart"', () => {
+    const locationSpy = jest.spyOn(window, 'location', 'get').mockReturnValue({
+      ...window.location,
```
Comment (jonkoops) [inline src/app/pages/MapPage/detail/buildDetailUrl.test.tsx:6]: You might get away without adding `window.location` here

File: src/app/pages/MapPage/detail/buildDetailUrl.test.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      type: 'foo',
+      subtype: 'bar',
+      id: 'baz',
+    }
+
+    const location = buildDetailUrl(params)
+
+    expect(location.pathname).toBe('/data/foo/bar/baz/')
+  })
+})
```
Comment (jonkoops) [inline src/app/pages/MapPage/detail/buildDetailUrl.test.tsx:35]: I'm missing a test case for the `kaarten` route.

**Review by jonkoops — APPROVED** [review]

File: src/app/pages/MapPage/detail/buildDetailUrl.test.tsx (hunk 2 above)
Comment (alimpens) [reply src/app/pages/MapPage/detail/buildDetailUrl.test.tsx:6]: You get a type error if you omit `window.location`, the `Location` type expects all `window.location` props. I could overwrite the type I guess, but the current approach might be cleaner?

File: src/app/pages/MapPage/detail/buildDetailUrl.test.tsx (hunk 3 above)
Comment (alimpens) [reply src/app/pages/MapPage/detail/buildDetailUrl.test.tsx:35]: Good catch, I'll add it

File: src/app/pages/MapPage/detail/buildDetailUrl.test.tsx (hunk 2 above)
Comment (jonkoops) [reply src/app/pages/MapPage/detail/buildDetailUrl.test.tsx:6]: If you want to pass only a subset of the fields in your type you can cast it using the `as` keyword. In this case that would be something like:

```ts
 const locationSpy = jest.spyOn(window, 'location', 'get').mockReturnValue({ pathname: '/kaart', search: '?foo=bar' } as Location)
```

Comment (alimpens) [reply src/app/pages/MapPage/detail/buildDetailUrl.test.tsx]: Nice, changed it

**Review by jonkoops — APPROVED** [review]

**Pull request merged**